### PR TITLE
fix(Core/GameObjects): Move activation code of traps to GO_ACTIVATED.…

### PR DESCRIPTION
**Pull request by UltraNix**
… Original author: @Gacko

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

None. Just improved logic.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -759,6 +739,35 @@ void GameObject::Update(uint32 diff)
                             }
                             else m_groupLootTimer -= diff;
```
Comment (Kitzunu) [inline src/server/game/Entities/GameObject/GameObject.cpp:740]: fallthrough to fix build

**Comment by UltraNix** [comment]
@Kitzunu Done.

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 2]
```diff
@@ -757,8 +737,40 @@ void GameObject::Update(uint32 diff)
                                 m_groupLootTimer = 0;
                                 lootingGroupLowGUID = 0;
                             }
-                            else m_groupLootTimer -= diff;
+                            else
+                            {
+                                m_groupLootTimer -= diff;
+                            }
+                        }
```
Comment (pklloveyou) [inline src/server/game/Entities/GameObject/GameObject.cpp:744]:
```
- }
+ break;
```
?.?

Comment (UltraNix) [reply src/server/game/Entities/GameObject/GameObject.cpp:744]: Good catch.

**Comment by FrancescoBorzi** [comment]
@UltraNix build failing

**Comment by UltraNix** [comment]
@FrancescoBorzi Done.

**Pull request merged**